File: app/src/main/java/com/example/macroz/myapplication/utils/ScreenUtils.java
package com.example.macroz.myapplication.utils;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.util.TypedValue;

import com.example.macroz.myapplication.mainactivity.BaseApplication;

/**
 *
 */
public class ScreenUtils {
    /**
     * 当前是否横屏
     *
     * @return
     */
    public static boolean isLandscape() {
        return BaseApplication.getInstance().getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
    }

    public static float dp2px(float dp) {
        return dp2px(BaseApplication.getInstance().getResources(), dp);
    }

    public static float dp2px(Resources resources, float dp) {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, resources.getDisplayMetrics());
    }

}

File: app/src/main/java/com/example/macroz/myapplication/utils/ScreenUtils.kt
package com.example.macroz.myapplication.utils

import android.content.res.Configuration
import android.content.res.Resources
import android.util.TypedValue

import com.example.macroz.myapplication.activity.BaseApplication

/**
 *
 */
class ScreenUtils {

    companion object {
        /**
         * 当前是否横屏
         *
         * @return
         */
        fun isLandscape(): Boolean {
            return BaseApplication.instance?.resources?.configuration?.orientation == Configuration.ORIENTATION_LANDSCAPE;
        }

        fun dp2px(dp: Float): Float {
            return dp2px(BaseApplication.instance?.resources, dp)
        }

        fun dp2px(resources: Resources?, dp: Float): Float {
            return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, resources?.displayMetrics)
        }
    }

}
